### PR TITLE
Use 3:3 for default search window

### PR DESCRIPTION
**Pull request by beomki-yeo**
Let's take a good number of search window by default

### CODE REVIEW

**Comment by niermann999** [comment]
Note though, that 3x3 is not a good default for any detector that is converted from ACTS. It only works well for the toy detector and wire chamber. For all other detectors 0x0 is correct

**Comment by krasznaa** [comment]
:thinking: Since I myself view the toy and wire chamber detectors as the testing ground, while the geometries coming from Acts (ODD, ITk) as the actual thing that we're trying to make to work, I'd vote for keeping the 0x0 window by default then...?

**Comment by beomki-yeo** [comment]
I worry that people (including myself) running the examples _naively_ will keep complaining on the bad tracking efficiency

**Comment by beomki-yeo** [comment]
BTW @niermann999  what does you mean with "it only works well toy and wire"? If we widen the search window, navigation with other detectors such as Itk or ODD will fail?

**Comment by niermann999** [comment]
> BTW @niermann999 what does you mean with "it only works well for toy and wire"? If we widen the search window, navigation with other detectors such as Itk or ODD will fail?

In general you are right, that should only lead to an explosion in the number of candidates to test and hence possibly to worse computing performance. However, many candidates will be tested twice and since there is no duplicate removal yet, I don't know what the result would be (the stepper might abort on 0 step size?). The ODD and ITk grids are at least not meant to be used like this, to my understanding.

I am also not sure about the navigation cache sizes, yet. That depends on how we will implement the calculation of the maximal number of candidates that come from a grid of these detectors (i.e. just the maximal bin size?) or whether we set a global maximum size and let the navigator handle the cache exhaustion automatically (which I think would be more robust, and I started to implement a while ago but haven't found the time to finish). On the host, this won't be a problem, but on device this could lead to weird effects

Bottom line, for the moment at least it is safer not to do a neighbourhood lookup on grids that contain the entire neighbourhood already in a single bin.

Maybe you can set the window size in the examples only if the detector is named "toy_detector" or "wire_chamber"?

**Comment by beomki-yeo** [comment]
Um... shouldn't be only one candiate picked up from the neighborhoods? I don't know why you are worrying about duplicate or explosion from this search window 🤔

**Comment by niermann999** [comment]
> Um... shouldn't be only one candiate picked up from the neighborhoods? I don't know why you are worrying about duplicate or explosion from this search window 🤔

No, all surface that intersect successfully with the initial ray are put into the cache and then in subsequent calls the navigator goes essentially into guided navigation through these candidates. The grids in the ODD and ITk, however, register a single surface multiple times into surrounding bins, so that when all of those bins would get included into the lookup, the same surface will be tested multiple times and multiple copies of it will be put into the cache. And since an entire neighbourhood is registered in a single bin, the number of entries per bin is much higher than for the toy detector and wire chamber, making a neighbourhood lookup on the ACTS exported detectors more expensive

**Comment by beomki-yeo** [comment]
OK I see. Thanks for the explanations!
